src/Bicep.Core/Registry/IArtifactReferenceFactory.cs

```diff
@@ -9,11 +9,13 @@
 
 namespace Bicep.Core.Registry;
 
-public interface IModuleReferenceFactory
+public interface IArtifactReferenceFactory
 {
     ImmutableArray<string> AvailableSchemes(Uri parentModuleUri);
 
+    ResultWithDiagnostic<ArtifactReference> TryGetArtifactReference(string reference, ArtifactType artifactType, Uri parentModuleUri);
+
     ResultWithDiagnostic<ArtifactReference> TryGetModuleReference(string reference, Uri parentModuleUri);
```

Comment:
Do we still need a module-specific method on this interface?

Feels like we could either:
- Refactor all usages to use the `TryGetArtifactReference()` method (and pass `ArtifactType.Module`) - preferable IMO if there aren't too many usages.
- Make this an extension method.